pilot/pkg/kube/inject/testdata/inject/auth.cert-dir.yaml.injected

```diff
@@ -113,7 +121,13 @@ spec:
         image: docker.io/istio/proxy_init:unittest
         imagePullPolicy: IfNotPresent
         name: istio-init
-        resources: {}
```

Comment:
Can you help to confirm that these diffs are due to the differences we have between helm sidecar configmap vs hardcoded template?

Reply:
Yes.
1. Resources for container `istio-init` are always empty in hardcoded template.
2. Env `ISTIO_META_CONFIG_NAMESPACE` and `ISTIO_METAJSON_LABELS` are not included in hardcoded template.
3. `envoyStatsd` can be disabled in sidecar configmap, while cannot be disabled in hardcoded one.
4. For `runAsUser: 1337`, it will be added in hardcoded template if InterceptionMode is `TPROXY`, different from which in sidecar configmap. 
```
// In hardcoded one
[[ if eq (annotation .ObjectMeta $interceptionModeKey .ProxyConfig.InterceptionMode) "TPROXY" -]]
capabilities:
  add:
  - NET_ADMIN
[[ end -]]
[[ if eq (annotation .ObjectMeta $interceptionModeKey .ProxyConfig.InterceptionMode) "TPROXY" -]]
runAsUser: 1337
{{- if and .SDSEnabled .EnableSdsTokenMount }}
runAsGroup: 1337
{{ end -}}
[[- end ]]

// In configmap
{{ "[[ if eq (annotation .ObjectMeta `sidecar.istio.io/interceptionMode` .ProxyConfig.InterceptionMode) \"TPROXY\" -]]" }}
capabilities:
add:
- NET_ADMIN
runAsGroup: 1337
{{ "[[ else -]]" }}
{{ if and .Values.global.sds.enabled .Values.global.sds.useTrustworthyJwt }}
runAsGroup: 1337
{{- end }}
runAsUser: 1337
{{ "[[- end ]]" }}
```

Reply:
Thanks for the explanation! Preserving behavior from configmap should be fine.

@mandarjog can you confirm this `runAsUser` can be replaced `runAsGroup` with no user surprise? I guess if they're already using atuo inject, it should be fine. Just want to make sure we don't miss some corner case.

Reply:
I double check the diff again, and now see another diff, `zipkinAddress=""`. Can you confirm that's expected as well?

Also @mandarjog @douglas-reid can one of you confirm an empty zipkinAddress is valid for `istio-proxy` container?

Reply:
The `zipkinAddress` won't be added if its value is empty in hardcoded template, while the insertion of `zipkinAddress` in sidecar configmap is controlled by tracer type.
Maybe i need to set a default value for `zipkinAddress` in UT?